### PR TITLE
Integrated CloudObject

### PR DESCRIPTION
**Pull request by JosepSampe**
This patch integrates the CloudObject class. 

To operate with CloudObjects, it adds 2 methods in internal_storage class  in order to put temporary data from a function and get this data from another function in a transparent way.

Writing **internal_storage** as a parameter of the function automatically creates an instance of it, allowing to execute the new `internal_storage.put_object()` and` internal_storage.get_object()` methods. The `put_object()` method returns a CloudObject instance.

Example:

```
import pywren_ibm_cloud as pywren

def my_function_put(text, internal_storage):
    co1 = internal_storage.put_object('Temp object test 1: {}'.format(text, ))
    co2 = internal_storage.put_object('Temp object test 2: {}'.format(text, ))

    return [co1, co2]

def my_function_get(co, internal_storage):
    data = internal_storage.get_object(co)

    return data

pw = pywren.ibm_cf_executor()
pw.call_async(my_function_put, 'Hello World')
cloudobjects = pw.get_result()
pw.map(my_function_get, cloudobjects)
print(pw.get_result())
pw.clean()
```
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by gilv** [comment]
@JosepSampe I still trying to estimate and test this change. just a quick remark...I think "internal_storage" is. not a good name, it's not internal and confusing.. may be we can change it to "pstorage" - like short of Persitent Storage ? I think it will be clear..

A general question on this change: Does this change backward compatible? So all what is worked in the past will continue work as is?

**Comment by JosepSampe** [comment]
@gilv Yes, this is pull request is backward compatible, in fact **internal_storage** parameter was there till the beginning as **ibm_cos**. I only added **put_object()** and **get_object()** methods.

I can put as a parameter whatever name ;) but I put these methods in **internal_storage** because the information that is returned through return statement of the function is stored with internal storage, and these additions stores temp data in the same way as return statement.

**Comment by gilv** [comment]
@JosepSampe yes,  internal_storage was there before, but it was sort of class with static methods, now it keep the names of COS object internal, right? in the past every signature had "biucket name, key, internal_storage"..now internal_storage contains bucket and key internally
Am I correct?

We need other name for the signature, more user friendly then "internal_storage"..like "pstorage" or perhaps something better name ;)

**Comment by JosepSampe** [comment]
@gilv No, in the past he had **internal_storage** and **storage_handler** as possible function args, and then we changed **storage_handler** to **ibm_cos**. 

**internal_storage** has all methods used by pywren to store serialized function and data, and all the necessary staff to run pywren, including the result of the function and the execution stats. Is for this that I put those new methods here.

We can change the name of **internal_storage** to simply **storage** or **storage_backend**, I don't know... ;)

**Comment by JosepSampe** [comment]
@gilv I can put those methods within ibm_cos(boto3) instance if you prefer, in such a way that instead of having :
```
internal_storage.put_object()
internal_storage.get_object()
```
We would have:

```
ibm_cos.put_tmp_object()
ibm_cos.get_tmp_object()
```

But this approach would not be generic...

File: examples/cloudobject.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def my_function_put(text, internal_storage):
+    co1 = internal_storage.put_object('Temp object test 1: {}'.format(text, ))
+    co2 = internal_storage.put_object('Temp object test 2: {}'.format(text, ))
+    return [co1, co2]
+
+
+def my_function_get(co, internal_storage):
+    data = internal_storage.get_object(co)
```
Comment (gilv) [inline examples/cloudobject.py]: @JosepSampe the "co" is actually an array of [co1, co2], right? ( this is what returned from my_function_put? but then my_function_get does get on the array? and get a single  variable "data" ?

Comment (JosepSampe) [reply examples/cloudobject.py]: @gilv Nop, "co" i only one CloudObject. In the example the first function returns an array of 2 CloudObjects [co1, co2]. This is the input of the 2nd function, This means that the map() spawns 2 functions, each one process one CloudObject.

Comment (gilv) [reply examples/cloudobject.py]: @JosepSampe right, my fault...

**Pull request merged**